### PR TITLE
elk improvements

### PR DESCRIPTION
**Pull request by ryane**
* improved documentation including how to customize install and removing ELK
* scheduler cpu and ram configuration now actually take effect
* more resilient kibana configuration for elasticsearch url
* updates terraform sample configs so that clouds are more likely capable of running ELK by default

### CODE REVIEW

**Comment by siddharthist** [comment]
Will test later today on DigitalOcean!

File: terraform/digitalocean.sample.tf [hunk 1]
```diff
@@ -13,6 +13,6 @@ module "do-hosts" {
   region_name = "nyc3" # this must be a region with metadata support
 
   control_count = 3
-  worker_count = 3
+  worker_count = 4
```
Comment (siddharthist) [inline terraform/digitalocean.sample.tf]: Is the increase from 3 to 4 worker nodes necessary? I've been able to run ELK with 3 just fine on DO, several times.

Comment (ryane) [inline terraform/digitalocean.sample.tf]: It's likely not strictly necessary on all clouds - it depends on a combination of the number of nodes and the image types (how much ram and cpu). I think 4 nodes covers it on all clouds I have tested so far and was just trying to keep things consistent.

**Comment by siddharthist** [comment]
All UIs working at their respective endpoints!

**Comment by ryane** [comment]
digitalocean build failed due to terraform issue (https://github.com/hashicorp/terraform/issues/5402)

**Pull request merged**